### ISSUE TITLE
KeyboardDatePicker accessibility issues - multiple buttons missing labels

### ISSUE THREAD

**Issue by alexbrandtalliance** [mui-org/material-ui-pickers#2111]
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui-pickers/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯
The KeyboardDatePicker calendar and month buttons are announced as 'unlabeled button" when using a screen reader like JAWS. A screen reader user has no context for the purpose of these buttons if they are unlabeled.

## Expected Behavior 🤔
All buttons should have an aria-label or title attribute for Assistive Technology users.

## Steps to Reproduce 🕹

https://qm22m.csb.app/
Steps:

Run thru the following steps with a screen reader like JAWS running. If you're not using a screen reader then inspect each button with the browser dev tools and confirm button is missing an aria-label or title attribute.

1. Press Tab to set focus to the Calendar icon button.
Result: Calendar Button announces as "Unlabeled Button"
2. While on the Calendar Button press enter to open the Calendar dialog.
3. Press Tab to set focus to the Previous Month button
Result: Previous Month button announces as "Unlabeled Button"
4. Press Tab to set focus to the Next Month button
Result: Next Month button announces as "Unlabeled Button"

## Context 🔦
The missing labels on buttons makes it hard for Assistive Technology users to use the Calendar dialog.

Running the Calendar Component thru Microsoft's Accessibility Insights tool provides the following failure details for the buttons.
---------------------------------------
Title: WCAG 4.1.2: Ensures buttons have discernible text (.MuiPickersCalendarHeader-iconButton.MuiIconButton-root:nth-child(3))
Tags: Accessibility, WCAG 4.1.2, button-name

Issue: Ensures buttons have discernible text (button-name - https://dequeuniversity.com/rules/axe/4.0/button-name?application=msftAI)

Target application: Datepicker - @material-ui/pickers component - https://material-ui-pickers.dev/demo/datepicker

Element path: .MuiDialogContent-root > .MuiPickersBasePicker-container > .MuiPickersBasePicker-pickerView > div:nth-child(1) > .MuiPickersCalendarHeader-switchHeader > .MuiPickersCalendarHeader-iconButton.MuiIconButton-root:nth-child(3)

Snippet: <button class="MuiButtonBase-root MuiIconButton-root MuiPickersCalendarHeader-iconButton" tabindex="0" type="button">

How to fix: 
Fix any of the following:
  Element does not have inner text that is visible to screen readers
  aria-label attribute does not exist or is empty
  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
  Element's default semantics were not overridden with role="presentation"
  Element's default semantics were not overridden with role="none"
  Element has no title attribute or the title attribute is empty

Environment: Microsoft Edge version 85.0.564.44

====

This accessibility issue was found using Accessibility Insights for Web 2.21.1 (axe-core 4.0.1), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.

Title: WCAG 4.1.2: Ensures buttons have discernible text (.MuiIconButton-root)
Tags: Accessibility, WCAG 4.1.2, button-name

Issue: Ensures buttons have discernible text (button-name - https://dequeuniversity.com/rules/axe/4.0/button-name?application=msftAI)

Target application: Datepicker - @material-ui/pickers component - https://material-ui-pickers.dev/demo/datepicker

Element path: .MuiFormControl-root.MuiTextField-root:nth-child(3) > .MuiInputBase-adornedEnd.MuiInput-root.MuiInput-underline > .MuiInputAdornment-positionEnd.MuiInputAdornment-root > .MuiIconButton-root

Snippet: <button class="MuiButtonBase-root MuiIconButton-root" tabindex="0" type="button">

How to fix: 
Fix any of the following:
  Element does not have inner text that is visible to screen readers
  aria-label attribute does not exist or is empty
  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
  Element's default semantics were not overridden with role="presentation"
  Element's default semantics were not overridden with role="none"
  Element has no title attribute or the title attribute is empty

Environment: Microsoft Edge version 85.0.564.44

====

This accessibility issue was found using Accessibility Insights for Web 2.21.1 (axe-core 4.0.1), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.

## Your Environment 🌎
This issue is not specific to our environment and occurs on the material-ui-pickers demo site. However, this issue has been reported by Assistive Technology users in our environment.

Screen shots of Accessibility Insights tool flagging buttons on the demo site.
![image](https://user-images.githubusercontent.com/22755642/93131860-904ae000-f689-11ea-8666-58ed9e77e74b.png)
![image](https://user-images.githubusercontent.com/22755642/93131878-96d95780-f689-11ea-91dd-99de0a1ce88a.png)

| Tech        | Version |
| ----------- | ------- |
| @material-ui/core | v4.11.0  |
| @material-ui/pickers | v3.2.9  |
| React       |         |
| Browser     |   Microsoft Edge Version 85.0.564.44, Chrome Version 85.0.4183.83   |
| TypeScript  |         |
| etc.        |         |

**Comment by oliviertassinari** [comment]
@alexbrandtalliance Could you try again with v4-alpha?

**Comment by alexbrandtalliance** [comment]
@oliviertassinari Tested v4-alpha and these issues are corrected. Do you know when these changes will move out of alpha?

**Comment by oliviertassinari** [comment]
> Do you know when these changes will move out of alpha?

I think between 6 and 12 months. Once we have nailed the component, we will make it stable. In the meantime, we need the flexibility to make breaking changes. From my experience so far, that's not what's stopping most developers to use a component.

**Comment by oliviertassinari** [comment]
Thanks for the confirmation on v4-alpha.
